lms/djangoapps/commerce/api/v1/models.py

```diff
@@ -0,0 +1,82 @@
+""" API v1 models. """
+from itertools import groupby
+import logging
+
+from django.db import transaction
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from course_modes.models import CourseMode
+
+log = logging.getLogger(__name__)
+
+
+class Course(object):
+    """ Pseudo-course model used to group CourseMode objects. """
+    id = None  # pylint: disable=invalid-name
+    modes = None
+
+    def __init__(self, id, modes):  # pylint: disable=invalid-name,redefined-builtin
+        self.id = CourseKey.from_string(unicode(id))  # pylint: disable=invalid-name
+        self.modes = list(modes)
+
+    @transaction.commit_on_success
```

Comment:
If there is an error while saving one of the models, we should not save any of them. This ensures atomicity.